src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -334,66 +395,98 @@ internal bool TryUnparse(StringBuilder sb)
             }
         }
 
-        public class ArgumentsCore
+        /// <summary>
+        /// The settings for <see cref="TextLoader"/>
+        /// </summary>
+        public class Options
         {
+            /// <summary>
+            /// Whether the input may include quoted values, which can contain separator characters, colons,
+            /// and distinguish empty values from missing values. When true, consecutive separators denote a
+            /// missing value and an empty value is denoted by \"\". When false, consecutive separators denote an empty value.
+            /// </summary>
             [Argument(ArgumentType.AtMostOnce,
                 HelpText =
                     "Whether the input may include quoted values, which can contain separator characters, colons," +
                     " and distinguish empty values from missing values. When true, consecutive separators denote a" +
                     " missing value and an empty value is denoted by \"\". When false, consecutive separators" +
                     " denote an empty value.",
                 ShortName = "quote")]
-            public bool AllowQuoting = DefaultArguments.AllowQuoting;
+            public bool AllowQuoting = Defaults.AllowQuoting;
 
+            /// <summary>
+            /// Whether the input may include sparse representations.
```

Comment:
>Whether the input may include sparse representations. [](start = 16, length = 53)

[citation needed] ([https://xkcd.com/285/](https://xkcd.com/285/))
not sure how it should look like, but we should specify how sparse format looks like. it's a index:value separated pairs where each index should be bigger than previous one.

Reply:
I also not sure can we have multiple sparse columns or not.

---
In reply to: [255806745](https://github.com/dotnet/machinelearning/pull/2417#discussion_r255806745) [](ancestors = 255806745)

Reply:
I would like to do another PR focusing on improving the public surface of text loader. I think these are all important points, and I will come back to them to improve the documentation. 
Before I focus on improving the documentation for these public fields I would like to make sure that we want to expose them all and that the current form is the final one.

---
In reply to: [255806866](https://github.com/dotnet/machinelearning/pull/2417#discussion_r255806866) [](ancestors = 255806866,255806745)